src/main/cpp/src/NativeParquetJni.cpp

```diff
@@ -221,21 +231,36 @@ class column_pruner {
       auto found       = children.find(name);
 
       if (found != children.end()) {
-        // found a match so update the number of children that passed the filter and ask it to
-        // filter itself.
-        ++schema_num_children[our_num_children_index];
+        // Record the current size of schema_map before processing the child
+        // This helps determine if the child was retained after processing.
+        std::size_t before_child_size = schema_map.size();
         found->second.filter_schema(schema,
                                     ignore_case,
                                     current_input_schema_index,
                                     next_input_chunk_index,
                                     chunk_map,
                                     schema_map,
                                     schema_num_children);
+
+        // If the size of schema_map has increased, it means that child was successfully
+        // included
+        if (schema_map.size() > before_child_size) {
+          ++matched_children_count;
+          // Update the child count for the struct
+          schema_num_children[struct_map_position]++;
+        }
       } else {
-        // No match was found so skip the child.
+        // No match was found so skip the child
```

Comment:
Absolute nit:  We are inconsistent with punctuation, in this change.  Note that line#235 has a period, but we opt to remove the period here.

My vote would be to punctuate deliberately, and add the period back.